test/FunctionalTests/Microsoft.OData.Core.Tests/ODataMultipartMixedBatchReaderTests.cs

```diff
@@ -205,6 +206,42 @@ public void ReadMultipartMixedBatchRequest()
             Assert.Empty(verifyResourceStack);
         }
 
+        [Fact]
+        public void ReadMultipartMixedBatchRequestWthoutContentId()
+        {
+            var payload = Regex.Replace(batchRequestPayload, "Content-ID: .", "");
+            Assert.NotEqual(payload, batchRequestPayload);
+            SetupMultipartMixedBatchReaderAndRunTest(
+                batchRequestPayload,
+                (multipartMixedBatchReader) =>
+                {
+                    var operationCount = 0;
+
+                    while (multipartMixedBatchReader.Read())
+                    {
+                        switch (multipartMixedBatchReader.State)
+                        {
+                            case ODataBatchReaderState.Operation:
+                                var operationRequestMessage = multipartMixedBatchReader.CreateOperationRequestMessage();
+                                using (var messageReader = new ODataMessageReader(operationRequestMessage, new ODataMessageReaderSettings(), this.model))
+                                {
+                                    if (operationCount == 3)

```

Comment:
At what point is the `operationCount` updated?

Reply:
Updated